llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -10035,7 +10035,7 @@ SDValue DAGCombiner::visitSHL(SDNode *N) {
   // Variant of version done on multiply, except mul by a power of 2 is turned
   // into a shift.
   if ((N0.getOpcode() == ISD::ADD || N0.getOpcode() == ISD::OR) &&
-      N0->hasOneUse() && TLI.isDesirableToCommuteWithShift(N, Level)) {
+      TLI.isDesirableToCommuteWithShift(N, Level)) {
```

Comment:
Simpler might be to make `TLI.isDesirableToComputeWithShift` return an enum s.t based on return its either:
1. Always preferable
2. Preferable if one use
3. Never preferable

Then you can just leave the 1-use check here as opposed to putting it in each backend.